var/spack/repos/builtin/packages/llvm-openmp/package.py

```diff
@@ -27,6 +27,11 @@ class LlvmOpenmp(CMakePackage):
         description="Support gomp and the Intel openMP runtime library.",
     )
 
+    requires("%clang", "%apple-clang", policy="one_of", msg="llvm-openmp builds only with Clang")
+    # Needs at least clang-16 to build @16:
+    conflicts("%clang@:15", when="@16")
+    conflicts("%apple-clang@:15", when="@16")
```

Comment:
The numbering scheme for Clang and Apple Clang doesn't match. The most recent version of Apple Clang is 14.0.3.